test-data/unit/check-dataclasses.test

```diff
@@ -665,3 +665,19 @@ def func() -> int: ...
 c: C
 c.x = 1  # E: Property "x" defined in "C" is read-only
 [builtins fixtures/bool.pyi]
+
+[case testTypeInDataclassDeferredStar]
+import lib
+[file lib.py]
+from dataclasses import dataclass
+from other import *
```

Comment:
Similar to above, maybe we should be able to process this import without adding anything to `missing_names`.

Reply:
Opened https://github.com/python/mypy/issues/7141 for these two.